packages/client/src/runtime/RequestHandler.ts

```diff
@@ -96,28 +99,30 @@ export class RequestHandler {
         const queries = requests.map((r) => r.protocolMessage.toEngineQuery())
         const traceparent = getTraceParent({ context: requests[0].otelParentCtx, tracingConfig: client._tracingConfig })
 
-        if (traceparent) info.headers.traceparent = traceparent
+        if (traceparent) info.traceparent = traceparent
         // TODO: pass the child information to QE for it to issue links to queries
         // const links = requests.map((r) => trace.getSpanContext(r.otelChildCtx!))
 
         const containsWrite = requests.some((r) => r.protocolMessage.isWrite())
 
-        const batchTransaction = info.transaction?.kind === 'batch' ? info.transaction : undefined
-
         return this.client._engine.requestBatch(queries, {
-          headers: info.headers,
-          transaction: batchTransaction,
+          traceparent,
+          transaction: getTransactionOptions(info.transaction),
           containsWrite,
+          customDataProxyHeaders: info.customDataProxyHeaders,
         })
       },
       singleLoader: (request) => {
         const info = getRequestInfo(request)
-        const interactiveTransaction = info.transaction?.kind === 'itx' ? info.transaction : undefined
+        const interactiveTransaction =
+          info.transaction?.kind === 'itx' ? getItxTransactionOptions(info.transaction) : undefined
```

Comment:
I think a batch with a single query is also possible, might be useful to set isolation on a `$queryRaw`